core/handler/device/store.go

```diff
@@ -23,11 +24,13 @@ type Store interface {
 	DownlinkQueue(appID, devID string) (DownlinkQueue, error)
 	Set(new *Device, properties ...string) (err error)
 	Delete(appID, devID string) error
+	AttrWhitelist() map[string]bool
 }
 
 const defaultRedisPrefix = "handler"
 const redisDevicePrefix = "device"
 const redisDownlinkQueuePrefix = "downlink"
+const redisUpStreamMetadataKeyWhitelist = ":attr-whitelist"
```

Comment:
1. Use the generic ID validator to check attribute keys
2. Restrict the length of attribute values to 200 bytes. Bit arbitrary but there should be a limit
3. There should be a list of builtin attributes, starting with `ttn-`, in configuration, whose set values should always be published upstream (e.g. `ttn-brand`, `ttn-model`). I think this is currently the "whitelist", but I'd call them "standard attributes", but open to suggestions
4. Other attributes should be limited by a fixed number, which is also set in the configuration. Which attributes are being published can be random or alphabetically sorted. The latter gives a consistent result